### ISSUE TITLE
envoy_reloadable_features_successful_active_health_check_uneject_host deprecation

### ISSUE THREAD

**Issue by alyssawilk** [envoyproxy/envoy#29099]
Your change #24331 (prefer active health check during outlier ejection wait) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

**Comment by suniltheta** [comment]
Can we get some more time before deprecating this flag? This is a behavior change that will affect if active health check is not same as the data plane traffic, i.e., if clients are doing shallow health checks on upstream clusters.

**Comment by alyssawilk** [comment]
@suniltheta It's totally possible to delay removal but it'd be helpful to have more information from  you to do that :-)  We've already had 7 months with this the default in Envoy, so live in production for many companies.  Are you running with it false locally and need more time to deploy it, or can you give more information why the usual process is problematic for this one?

**Comment by suniltheta** [comment]
Yes, currently running with flag as false given this slight change in behavior can impact cases where active health check is not reflecting the data plane traffic.

In the original issue https://github.com/envoyproxy/envoy/issues/24217 it was considered whether to have a config option for enabling this behavior change but was done with runtime flag.

I agree that 7 months/2 releases is sufficient enough time. If I am the only one, we can go ahead with removing this flag.

**Comment by alyssawilk** [comment]
Gotcha.  Would the slight change be problematic for you folks?   If you can canary and test and have problems for your set up I think it'd be fine to add a permanent knob.  Happy to give you an extra month or two grace time for testing if that'd help out

**Comment by alyssawilk** [comment]
testing would be great - if there are problems it'd be good to know for others as well.  If all else fails and you don't have the cycle to debug, it'd be fantastic if you could help add a permanent knob :-)

**Comment by suniltheta** [comment]
New PR https://github.com/envoyproxy/envoy/pull/29303 will deprecated the runtime flag envoy_reloadable_features_successful_active_health_check_uneject_host and adds a config option to set it via Outlier Detection config.

**Comment by suniltheta** [comment]
From my testing there is no issue with unexpected stats like I originally thought. It was the issue with my testing where intentionally simulating 5xx error was retried multiple times from the http client that retry on error.

**Comment by alyssawilk** [comment]
Thank you so much for testing this and verifying it was not problematic.  We really appreciate being able to clean up the legacy code paths!
